### ISSUE TITLE
ERROR [2/2] RUN apt-get update &&     apt-get -y install git &&

### ISSUE THREAD

**Issue by dev-naddari** [Rikj000/MoniGoMani#4]
Hello 
I had an error during installation, any hint please:

docker-compose pull
Pulling freqtrade ... done 

docker-compose up --build
Building freqtrade
failed to get console mode for stdout: The handle is invalid.
[+] Building 1.0s (4/5)
 => [internal] load build definition from Dockerfile.technical             0.0s
[+] Building 1.1s (5/5) FINISHED
 => [internal] load build definition from Dockerfile.technical             0.0s
 => => transferring dockerfile: 42B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/freqtradeorg/freqtrade:develop  0.0s
 => [1/2] FROM docker.io/freqtradeorg/freqtrade:develop                    0.1st &&     apt-get clean &&     pip install
 **=> ERROR [2/2] RUN apt-get update &&     apt-get -y install git &&**     a  0.8s
------
 > [2/2] RUN apt-get update &&     apt-get -y install git &&     apt-get clean &&     pip install git+https://github.com/freqtrade/technical:
#5 0.224 Reading package lists...
#5 0.816 E: Could not open lock file /var/lib/apt/lists/lock - open (13: Permission denied)
#5 0.816 E: Unable to lock directory /var/lib/apt/lists/
------
ERROR: Service 'freqtrade' failed to build

**Comment by Rikj000** [comment]
Correct, I experienced this issue today too, I resolved it by doing the following 2 things:   
-[Download](https://github.com/freqtrade/freqtrade/tree/develop/docker) & the latest official Freqtrade `docker` folder & `Dockerfile.`'s that are within & replace the ones in `MoniGoMani/docker/` with these.   
-Update your `docker-compose.yml` as following:
```yml
---
version: '3'
services:
  freqtrade:
    # image: freqtradeorg/freqtrade:stable
    image: freqtradeorg/freqtrade:develop
    # Use plotting image
    # image: freqtradeorg/freqtrade:develop_plot
    # Build step - only needed when additional dependencies are needed
    # build:
    #  context: .
    #  dockerfile: "./docker/Dockerfile.technical"
    restart: unless-stopped
    container_name: Freqtrade-MoniGoMani
    volumes:
      - "/etc/localtime:/etc/localtime:ro"
      - "./user_data:/freqtrade/user_data"
    # Expose api on port 8080 (localhost only)
    # Please read the https://www.freqtrade.io/en/latest/rest-api/ documentation
    # before enabling this.
    ports:
      - "127.0.0.1:8080:8080"
    # Default command used when running `docker compose up`
    command: >
      trade
      --logfile /freqtrade/user_data/logs/freqtrade.log
      --db-url sqlite:////freqtrade/user_data/tradesv3-MoniGoMani_v0.8.1.sqlite
      --config /freqtrade/user_data/config.json
      --config /freqtrade/user_data/config-private.json
      --strategy MoniGoManiHyperStrategy
```
- Stop & delete your current `Freqtrade-MoniGoMani`container   
- Do `docker-compose pull` & `docker-compose --build --force-recreate`   
   
After that your problem should be resolved, these fixes will be included in **v0.9.0** once it is complete.

**Comment by dev-naddari** [comment]
hello Rikj000, thanks a lot, I did Above, error above has gone but I am receiving new one (log file not found), see the screen dump:
docker-compose up --build
Creating Freqtrade-MoniGoMani ... done
Attaching to Freqtrade-MoniGoMani
Freqtrade-MoniGoMani | 2021-04-10 21:47:19,548 - freqtrade.worker - INFO - Starting worker docker-aaf9872e
Freqtrade-MoniGoMani | 2021-04-10 21:47:19,548 - freqtrade.configuration.configuration - INFO - Using config: /freqtrade/user_data/config.json ...
Freqtrade-MoniGoMani | 2021-04-10 21:47:19,556 - freqtrade.configuration.configuration - INFO - Using config: /freqtrade/user_data/config-private.json ...
Freqtrade-MoniGoMani | 2021-04-10 21:47:19,559 - freqtrade.commands.trade_commands - ERROR - [Errno 2] No such file or directory: '/freqtrade/user_data/logs/freqtrade.log'
Freqtrade-MoniGoMani | 2021-04-10 21:47:19,559 - freqtrade.commands.trade_commands - ERROR - Fatal exception!
Freqtrade-MoniGoMani | Traceback (most recent call last):
Freqtrade-MoniGoMani |   File "/freqtrade/freqtrade/commands/trade_commands.py", line 18, in start_trading
Freqtrade-MoniGoMani |     worker = Worker(args)
Freqtrade-MoniGoMani |   File "/freqtrade/freqtrade/worker.py", line 35, in __init__
Freqtrade-MoniGoMani |     self._init(False)
Freqtrade-MoniGoMani |   File "/freqtrade/freqtrade/worker.py", line 49, in _init
Freqtrade-MoniGoMani |     self._config = Configuration(self._args, None).get_config()
Freqtrade-MoniGoMani |   File "/freqtrade/freqtrade/configuration/configuration.py", line 41, in get_config
Freqtrade-MoniGoMani |     self.config = self.load_config()
Freqtrade-MoniGoMani |   File "/freqtrade/freqtrade/configuration/configuration.py", line 99, in load_config
Freqtrade-MoniGoMani |     self._process_logging_options(config)
Freqtrade-MoniGoMani |   File "/freqtrade/freqtrade/configuration/configuration.py", line 131, in _process_logging_options
Freqtrade-MoniGoMani |     setup_logging(config)
Freqtrade-MoniGoMani |   File "/freqtrade/freqtrade/loggers.py", line 109, in setup_logging
Freqtrade-MoniGoMani |     handler_rf = RotatingFileHandler(logfile,
Freqtrade-MoniGoMani |   File "/usr/local/lib/python3.9/logging/handlers.py", line 153, in __init__
Freqtrade-MoniGoMani |     BaseRotatingHandler.__init__(self, filename, mode, encoding=encoding,
Freqtrade-MoniGoMani |   File "/usr/local/lib/python3.9/logging/handlers.py", line 58, in __init__
Freqtrade-MoniGoMani |     logging.FileHandler.__init__(self, filename, mode=mode,
Freqtrade-MoniGoMani |   File "/usr/local/lib/python3.9/logging/__init__.py", line 1146, in __init__
Freqtrade-MoniGoMani |     StreamHandler.__init__(self, self._open())
Freqtrade-MoniGoMani |   File "/usr/local/lib/python3.9/logging/__init__.py", line 1175, in _open
Freqtrade-MoniGoMani |     return open(self.baseFilename, self.mode, encoding=self.encoding,
Freqtrade-MoniGoMani | FileNotFoundError: [Errno 2] No such file or directory: '/freqtrade/user_data/logs/freqtrade.log'

**Comment by dev-naddari** [comment]
It is an access right issue ( I dont know how to elevate the privileges ) so I changed the path to the logfile in docker-compose.yml to  --logfile /tmp/freqtrade.log
And this solved the issue, the bit is up an drunning now

**Comment by Rikj000** [comment]
Your 2nd issue should already have been resolved with [this commit](https://github.com/Rikj000/MoniGoMani/commit/91fe0d24e88eff89f82478cf842afbdafca75a2d) on the `main` branch of MoniGoMani & the fix will also be included in **v0.9.0** when it comes out.
All that should be needed to resolve your 2nd issue is creating an empty `MoniGoMani/user_data/logs/freqtrade.log`file.

**Comment by dev-naddari** [comment]
Brilliant, thanks Rikj000
